org.eclipse.xtext.xbase/src/org/eclipse/xtext/xbase/typesystem/computation/CollectionLiteralsTypeComputer.java

```diff
@@ -176,7 +177,17 @@ protected void computeType(XListLiteral literal, JvmGenericType listType, ITypeE
 
 	protected void computeArrayLiteralType(XListLiteral literal, LightweightTypeReference expectedArrayType, ITypeExpectation expectation,
 			ITypeComputationState state) {
-		LightweightTypeReference elementTypeExpectation = expectedArrayType.getComponentType();
+		LightweightTypeReference elementTypeExpectation = null;
```

Comment:
Could it be also that CompoundTypeReference.getComponentType for the case when CompoundTypeReference.isArray returns true is bogus?